pip/index.py

```diff
@@ -640,6 +642,12 @@ def _link_package_versions(self, link, search):
                 self._log_skipped_link(
                     link, 'Python version is incorrect')
                 return
+
+        if not check_requires_python(link.requires_python):
```

Comment:
Good point, I went for accepting the link, as ignoring can be hard to debug on end-user side.